include/sys/util.h

```diff
@@ -106,7 +106,7 @@ extern "C" {
  * In C, passing a pointer as @p array causes a compile error.
  */
 #define ARRAY_SIZE(array) \
-	((long) (IS_ARRAY(array) + (sizeof(array) / sizeof((array)[0]))))
+	((size_t) (IS_ARRAY(array) + (sizeof(array) / sizeof((array)[0]))))
```

Comment:
The type of `size_t / size_t` is already implicitly `size_t`, so the explicit cast isn't needed. Omitting the explicit cast also makes this consistent with the other `ARRAY_SIZE` macro definition in this header file (that's used for C++ code).

Reply:
@Thalley right, that's a fair point. I guess the explicit cast can't be avoided then.